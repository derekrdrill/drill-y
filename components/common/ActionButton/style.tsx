import styled from 'styled-components';
import { Button } from '@mui/material';

export const ActionButtonRoot = styled(Button)<{
  $borderRadius?: number;
  $buttonColor?: string;
  $buttonHoverColor?: string;
  $buttonWidth?: string | number;
  $marginTop?: number;
  disabled?: boolean;
  variant?: string;
}>(
  ({
    $borderRadius,
    $buttonColor,
    $buttonHoverColor,
    $buttonWidth,
    $marginTop,
    disabled,
    variant,
  }) => ({
    '&:hover': {
      backgroundColor: `${$buttonHoverColor} !important`,
      color: `#FFFFFF !important`,
    },
    backgroundColor: `${variant === 'contained' ? `${$buttonColor}` : 'inherit'} !important`,
    borderColor: disabled
      ? 'gainsboro'
      : `${variant === 'outlined' ? `${$buttonColor}` : 'inherit'} !important`,
    borderRadius: `${$borderRadius ?? 75}px !important`,
    boxShadow: 'none !important',
    color: disabled
      ? 'gainsboro'
      : `${variant === 'outlined' ? `${$buttonColor}` : '#FFFFFF'} !important`,
    fontFamily: `'Josefin Sans', sans-serif !important`,
    fontWeight: '600 !important',
    fontSize: '12px !important',
    marginTop: `${$marginTop ?? 4}px !important`,
    paddingBottom: '4px !important',
    paddingLeft: '20px !important',
    paddingRight: '20px !important',
    paddingTop: '6px !important',
    width: $buttonWidth
      ? `${typeof $buttonWidth === `number` ? `${$buttonWidth}px` : $buttonWidth} !important`
      : 'inherit',
  }),
);
