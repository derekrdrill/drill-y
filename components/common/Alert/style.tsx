import styled from 'styled-components';
import { Grid } from '@mui/material';

// export const AlertContainer = styled(Grid)<{ isAlertHidden?: boolean; isAlertFading?: boolean }>(
//   ({ isAlertHidden, isAlertFading }) => ({
//     display: isAlertHidden && `none !important`,
//     opacity: isAlertHidden || isAlertFading ? 0 : 1,
//     position: 'fixed',
//     top: 125,
//     transition: 'all 350ms ease-in',
//     zIndex: 6,
//   }),
// );

export const AlertContainer = styled(Grid)<{
  $isAlertHidden?: boolean | undefined;
  $isAlertFading?: boolean | undefined;
}>(({ $isAlertHidden, $isAlertFading }) => ({
  display: `${$isAlertHidden ? 'none' : 'inline-flex'} !important`,
  opacity: $isAlertHidden || $isAlertFading ? 0 : 1,
  position: 'fixed',
  top: 125,
  transition: 'all 350ms ease-in',
  zIndex: 6,
}));
