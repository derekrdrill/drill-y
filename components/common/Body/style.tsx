import styled from 'styled-components';
import { Grid } from '@mui/material';

export const BodyRoot = styled(Grid)({
  zIndex: 0,
});
