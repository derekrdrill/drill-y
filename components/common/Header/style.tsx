import styled from 'styled-components';
import { AppBar, Grid } from '@mui/material';
import SearchIcon from '@mui/icons-material/Search';
import ShoppingBagIcon from '@mui/icons-material/ShoppingBag';

export const HeaderRoot = styled(AppBar)({
  backgroundColor: 'white !important',
  borderBottom: '1px solid #DFDFDF',
  boxShadow: 'none !important',
  zIndex: '3 !important',
});

export const HeaderTop = styled(Grid)({
  borderBottom: '1px solid #DFDFDF',
  padding: '5px 0',
});

export const HeaderLogo = styled.img({
  '&:hover': {
    opacity: 0.8,
  },
  cursor: 'pointer',
  height: 70,
  width: 100,
});

export const HeaderMenuText = styled.h4({
  '&:hover': {
    backgroundPositionX: '0%',
    backgroundSize: '100% 0.1em',
  },
  backgroundImage: 'linear-gradient(#212121, #212121)',
  backgroundPositionX: '100%',
  backgroundPositionY: '100%',
  backgroundRepeat: 'no-repeat',
  backgroundSize: '0% 0.1em',
  color: 'black',
  cursor: 'pointer',
  fontFamily: `'Josefin Sans', sans-serif !important`,
  fontWeight: '600 !important',
  textDecoration: 'none',
  textTransform: 'uppercase',
  transition: 'background-size 0.2s ease-in-out',
});

export const HeaderSearchIcon = styled(SearchIcon)({
  '&:hover': {
    color: '#a501c6',
    transform: 'scale(1.6)',
  },
  cursor: 'pointer',
  marginTop: 7,
  transition: 'all 5s ease-in-out',
  transform: 'scale(1.4)',
});
