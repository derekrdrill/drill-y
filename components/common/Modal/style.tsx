import styled from 'styled-components';
import { Grid } from '@mui/material';

export const ModalRootContainer = styled(Grid)({
  position: 'absolute',
  top: 200,
});

export const ModalContainer = styled(Grid)({
  backgroundColor: '#FFFFFF !important',
  borderRadius: 7,
  padding: '10px 20px',
});

export const ModalRow = styled(Grid)<{ isBottom?: boolean; isBody?: boolean }>(
  ({ isBottom, isBody }) => ({
    borderBottom: !isBottom ? '1px gainsboro solid' : 'none',
    margin: !isBottom ? '10px 0' : 0,
    padding: isBody ? '20px 0' : '5px 0',
  }),
);
