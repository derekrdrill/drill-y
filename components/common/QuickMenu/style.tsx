import styled from 'styled-components';
import { Grid } from '@mui/material';

export const QuickMenuRoot = styled(Grid)<{ $isQuickMenuOpen?: boolean }>(
  ({ $isQuickMenuOpen }) => ({
    backgroundColor: '#FFFFFF',
    boxShadow: '0 4px 2px -2px gray',
    left: 0,
    height: 400,
    minWidth: '100%',
    position: 'fixed',
    top: 123,
    transform: $isQuickMenuOpen ? 'none' : 'translateY(-105%)',
    transition: 'transform 300ms linear',
    width: '100%',
    zIndex: 2,
  }),
);
