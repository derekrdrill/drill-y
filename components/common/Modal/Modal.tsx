import * as React from 'react';
import { Button, Grid, Modal as ModalComponent, Typography } from '@mui/material';

import GlobalContext from '../../../context/GlobalContext';

import { setModalItem } from './actions/ModalActions';

import { ModalContainer, ModalRootContainer, ModalRow } from './style';

const Modal = () => {
  const {
    state: {
      modalItem: {
        isModalOpen,
        modalBody,
        modalTitle,
        handleSubmit,
        submitButtonColor,
        submitButtonText,
        isCancelHidden,
        submitButtonVariant,
      },
    },
    dispatch,
  } = React.useContext(GlobalContext);

  return (
    <ModalComponent open={isModalOpen}>
      <ModalRootContainer container>
        <Grid item xs={3} />
        <Grid item xs={6}>
          <ModalContainer container>
            <Grid item xs={12}>
              <ModalRow container>
                <Grid item xs={11}>
                  <Typography variant='h6'>{modalTitle}</Typography>
                </Grid>
                <Grid item xs={1}>
                  <Grid container justifyContent='flex-end'>
                    {!isCancelHidden && (
                      <Button
                        color='error'
                        variant='contained'
                        onClick={
                          /* istanbul ignore next */
                          () => dispatch(setModalItem(null, false, null, ''))
                        }
                        size='small'
                      >
                        x
                      </Button>
                    )}
                  </Grid>
                </Grid>
              </ModalRow>
              <ModalRow container isBody>
                <Grid item xs={12}>
                  {modalBody}
                </Grid>
              </ModalRow>
              <ModalRow columnSpacing={2} container isBottom justifyContent='flex-end'>
                <Grid item>
                  {!isCancelHidden && (
                    <Button
                      color='info'
                      variant='contained'
                      onClick={
                        /* istanbul ignore next */
                        () => dispatch(setModalItem(null, false, null, ''))
                      }
                    >
                      Cancel
                    </Button>
                  )}
                </Grid>
                <Grid item>
                  <Button
                    color={submitButtonColor}
                    variant={submitButtonVariant}
                    onClick={
                      /* istanbul ignore next */
                      () => handleSubmit()
                    }
                  >
                    {submitButtonText}
                  </Button>
                </Grid>
              </ModalRow>
            </Grid>
          </ModalContainer>
        </Grid>
        <Grid item xs={3} />
      </ModalRootContainer>
    </ModalComponent>
  );
};

export default Modal;
