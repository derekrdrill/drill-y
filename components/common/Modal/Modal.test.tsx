import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Modal from './Modal';

import { ModalRow } from './style';

describe('Modal unit tests', () => {
  it('renders as expected', () => {
    const result = renderer.create(<Modal />).toJSON();
    expect(result).toMatchSnapshot();
  });

  it('renders ModalRow correctly (isBottom)', async () => {
    render(<ModalRow data-testid='modal-row' isBottom />);

    const modalRow = await screen.findAllByTestId('modal-row');

    expect(modalRow[0]).toHaveClass('sc-dmqHEX eOaCfP');
  });

  it('renders ModalRow correctly (isBody)', async () => {
    render(<ModalRow data-testid='modal-row' isBody />);

    const modalRow = await screen.findAllByTestId('modal-row');

    expect(modalRow[0]).toHaveClass('sc-dmqHEX kDoKht');
  });
});
