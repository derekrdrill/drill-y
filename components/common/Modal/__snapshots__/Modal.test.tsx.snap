// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal unit tests renders as expected 1`] = `null`;
